controllers/platformnetwork_controller.go

```diff
@@ -661,37 +739,180 @@ func (r *PlatformNetworkReconciler) statusUpdateRequired(instance *starlingxv1.P
 	return !instance.Status.DeepEqual(status)
 }
 
-// ReconcileResource interacts with the system API in order to reconcile the
-// state of a data network with the state stored in the k8s database.
-func (r *PlatformNetworkReconciler) ReconcileResource(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (err error) {
-	oldStatus := instance.Status.DeepCopy()
+// Determine if the system type is All-in-one simplex, return the Host object if so.
+func (r *PlatformNetworkReconciler) AIOSXHost(client *gophercloud.ServiceClient) (*gcHosts.Host, bool, error) {
+	all_hosts, err := gcHosts.ListHosts(client)
+	if err != nil {
+		return nil, false, common.NewSystemDependency("Failed to get host list")
+	}
 
-	if instance.DeletionTimestamp.IsZero() {
-		err = r.ReconcileAddressPool(client, instance)
-		if err == nil {
-			if networkResourceRequired(instance) {
-				err = r.ReconcileNetwork(client, instance)
+	if len(all_hosts) == 1 {
+		for _, host := range all_hosts {
+			if host.Personality == "controller" {
+				default_system, err := gcSystem.GetDefaultSystem(client)
+				if err != nil {
+					return nil, false, common.NewSystemDependency("Failed to get default system")
+				}
+
+				if default_system.SystemMode == "simplex" && default_system.SystemType == "All-in-one" {
+					return &host, true, nil
+				}
 			}
 		}
+	}
 
-		inSync := err == nil
+	return nil, false, nil
+}
 
-		if inSync != instance.Status.InSync {
-			r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "synchronization has changed to: %t", inSync)
-		}
+func (r *PlatformNetworkReconciler) AddressPoolUpdateRequired(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (bool, error) {
+	var pool *addresspools.AddressPool
+	var err error
+	if instance.Spec.Type != networks.NetworkTypeOther {
+		pool, err = r.FindAddressPoolByNetworkType(client, instance)
+	} else {
+		pool, err = r.FindExistingAddressPool(client, instance)
+	}
+	if err != nil {
+		return false, err
+	} else if pool == nil {
+		return true, nil
+	}
```

Comment:
We are not returning pool over there, it's just the err if there is any.
Since previous condition evaluates err != nil, in the `else if pool == nil` block we can be sure that there is no err and hence we are returning nil.